wardrobe/tradable_kitties/src/lib.rs

```diff
@@ -0,0 +1,446 @@
+    return Ok(0);
+}
+
+/// Wrapper function for checking conversion from  tradable kitty to basic kitty.
+/// Multiple kitties can be converted from tradable to non-tradable in the same txn.
+fn check_can_delist_kitty_from_sale(
+    input_data: &[DynamicallyTypedData],
+    output_data: &[DynamicallyTypedData],
+) -> Result<TransactionPriority, TradeableKittyError> {
+    // Below is conversion from tradable kitty to kitty, reverse of the ListKittyForSale,
+    // hence input params are rebversed
+    check_kitty_tdkitty_interconversion(&output_data, &input_data)?;
+    return Ok(0);
+}
+
+/// Validaes inter-conversion b/w both kitty & tradable kitty.Used by listForSale & delistFromSale functions.
+fn check_kitty_tdkitty_interconversion(
+    kitty_data: &[DynamicallyTypedData],
+    tradable_kitty_data: &[DynamicallyTypedData],
+) -> Result<TransactionPriority, TradeableKittyError> {
+    ensure!(
+        kitty_data.len() == tradable_kitty_data.len() && !kitty_data.is_empty(),
+        { TradeableKittyError::NumberOfInputOutputMismatch }
+    );
+
+    let mut dna_to_tdkitty_set: BTreeSet<KittyDNA> = BTreeSet::new();
+    let mut dna_to_kitty_set: BTreeSet<KittyDNA> = BTreeSet::new();
+
+    for i in 0..kitty_data.len() {
+        let utxo_kitty = kitty_data[i]
+            .extract::<KittyData>()
+            .map_err(|_| TradeableKittyError::BadlyTyped)?;
+
+        if dna_to_kitty_set.contains(&utxo_kitty.dna) {
+            return Err(TradeableKittyError::DuplicateKittyFound);
+        } else {
+            dna_to_kitty_set.insert(utxo_kitty.clone().dna);
+        }
+
+        let utxo_tradable_kitty = tradable_kitty_data[i]
+            .extract::<TradableKittyData>()
+            .map_err(|_| TradeableKittyError::BadlyTyped)?;
+
+        if dna_to_tdkitty_set.contains(&utxo_tradable_kitty.kitty_basic_data.dna) {
+            return Err(TradeableKittyError::DuplicateTradableKittyFound);
+        } else {
+            dna_to_tdkitty_set.insert(utxo_tradable_kitty.clone().kitty_basic_data.dna);
+        }
+
+        ensure!(
+            utxo_kitty == utxo_tradable_kitty.kitty_basic_data,
+            TradeableKittyError::KittyBasicPropertiesAltered
+        );
+        ensure!(
+            utxo_tradable_kitty.price != None, // basic kitty data is unaltered
+            TradeableKittyError::KittyPriceCantBeNone  // this need to be chan
+        );
+    }
+
+    return Ok(0);
+}
+
+impl<const ID: u8> SimpleConstraintChecker for TradableKittyConstraintChecker<ID> {
+    type Error = TradeableKittyError;
+
+    fn check(
+        &self,
+        input_data: &[DynamicallyTypedData],
+        _peeks: &[DynamicallyTypedData],
+        output_data: &[DynamicallyTypedData],
+    ) -> Result<TransactionPriority, Self::Error> {
+        match &self {
+            Self::ListKittyForSale => {
+                check_can_list_kitty_for_sale(&input_data, &output_data)?;
+                return Ok(0);
```

Comment:
You don't need to return early here (unless you want to specify a dedicated priority).

Reply:
fixed this.